mathics/builtin/base.py

```diff
@@ -721,11 +721,17 @@ def boxes_to_tex(self, leaves, **options) -> str:
 class PatternError(Exception):
     def __init__(self, name, tag, *args):
         super().__init__()
+        self.name = name
+        self.tag = tag
+        self.args = args
 
 
 class PatternArgumentError(PatternError):
     def __init__(self, name, count, expected):
         super().__init__(None, None)
```

Comment:
If instead of `super().__init__(None, None)` you change to `super().__init__(name, count)` then you don't have to set these explicitly in the next two lines below.